lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
+    rule.imply(chunk.rule);
+
+    visit(node.strings.first);
+    var previous = node.strings.first;
+    for (var string in node.strings.skip(1)) {
+      // Always split after a string that ends in a newline. Otherwise, indent a
+      // little bit so it's clear that the next string is a continuation rather
+      // than a new line.
+      if (_endsWithNewline(previous)) {
+        builder.writeWhitespace(Whitespace.nestedNewline);
+      } else {
+        builder.indent();
+        split();
+        builder.unindent();
+      }
```

Comment:
"Nesting" also causes indentation. The formatter internally uses "indent" to mean the sort of "block-level" indentation you get from statements and "nesting" to refer to the indentation you get when line wraps appear inside expressions.

A lot of the code assumes there will never be nesting in play at the point when indentation changes, because that implies you're still in the middle of an expression when crossing a statement boundary, which shouldn't be the case.

So instead of using `indent()` and `unindent()` here, you'll use `nestExpression()` and `unnest()`.